### DISCUSSION TITLE
src/Kernel.php, why is it defined in routing configuration

### DISCUSSION THREAD

**Discussion by lanrenbulan** [symfony/symfony#44784] [Q&A] [answered]
Hi,

I don't understand why there is a kernel in this configuration.

```# config/routes/annotations.yaml
controllers:
    resource: ../../src/Controller/
    type: annotation

kernel:
    resource: ../../src/Kernel.php
    type: annotation```


I found the same question:  [why-symfony-generates-route-with-kernel-name](https://stackoverflow.com/questions/66349579/why-symfony-generates-route-with-kernel-name)

**Comment by ahundiak** [comment, marked as answer]
As the comment on the question you linked implies, the Kernel is actually quite a powerful class that can do quite a bit of surprising stuff.  If, for example, you only have one or two action methods then you can just add them to the Kernel along with the routing annotation.  No real need for a controller class.  Not something you would normally do but it's there if you need it.

Also, as the linked question suggests, you can just comment out the annotation if you don't use it.  I think it's one of those things that are left in there to prevent BC breakage.
